lib/smart_proxy_remote_execution_ssh/cockpit.rb

```diff
@@ -45,9 +45,9 @@ def self.applies_for?(socket)
       end
     end
 
-    class OpenSSLBufferedSocket < BufferedSocket
+    class SSLBufferedSocket < BufferedSocket
       def self.applies_for?(socket)
-        socket.is_a? ::OpenSSL::SSL::SSLSocket
+        socket.is_a? ::OpenSSL::SSL::SSLSocket or socket.is_a? ::Puma::MiniSSL::Socket
```

Comment:
Not tested yet, but just in case Puma is not installed:

```ruby
return true if socket.is_a? ::OpenSSL::SSL::SSLSocket 
return true if defined?  ::Puma::MiniSSL::Socket && socket.is_a? ::Puma::MiniSSL::Socket
```